framework/src/modules/dpos_v2/internal_method.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { MethodContext } from '../../state_machine';
+import { BaseMethod } from '../base_method';
+import { NamedRegistry } from '../named_registry';
+import { MAX_NUMBER_BYTES_Q96 } from './constants';
+import { RewardsAssignedEvent } from './events/rewards_assigned';
+import { DelegateAccount, TokenMethod, VoterData } from './types';
+import { calculateVoteRewards } from './utils';
+
+export class InternalMethod extends BaseMethod {
+	private _tokenMethod!: TokenMethod;
+	private readonly _moduleName!: string;
+
+	public constructor(stores: NamedRegistry, events: NamedRegistry, moduleName: string) {
+		super(stores, events);
+
+		this._moduleName = moduleName;
+	}
+	public addDependencies(tokenMethod: TokenMethod) {
+		this._tokenMethod = tokenMethod;
+	}
+
+	public async assignVoteRewards(
+		methodContext: MethodContext,
+		voterAddress: Buffer,
+		voterData: VoterData,
+		indexWithDelegateVote: number,
```

Comment:
Maybe instead of these 2 inputs, receive `VoteObject` and return updated `VoteObject` (updated sharing coefficient) would be better. it would avoid unclear mutation and index